DllImportGenerator/Ancillary.Interop/SpanMarshallers.cs

```diff
@@ -0,0 +1,398 @@
+        }
+
+        public SpanMarshaller(Span<T> managed, Span<byte> stackSpace, int sizeOfNativeElement)
+        {
+            inner = new ReadOnlySpanMarshaller<T>(managed, stackSpace, sizeOfNativeElement);
+        }
+
+        /// <summary>
+        /// Stack-alloc threshold set to 256 bytes to enable small arrays to be passed on the stack.
+        /// Number kept small to ensure that P/Invokes with a lot of array parameters doesn't
+        /// blow the stack since this is a new optimization in the code-generated interop.
+        /// </summary>
+        public const int StackBufferSize = ReadOnlySpanMarshaller<T>.StackBufferSize;
+
+        public Span<T> ManagedValues => inner.ManagedValues;
+
+        public Span<byte> NativeValueStorage
+        {
+            get => inner.NativeValueStorage;
+        }
+
+        public ref byte GetPinnableReference() => ref inner.GetPinnableReference();
+
+        public void SetUnmarshalledCollectionLength(int length)
+        {
+            inner.SetUnmarshalledCollectionLength(length);
+        }
+
+        public byte* Value
+        {
+            get => inner.Value;
+            set => inner.Value = value;
+        }
+
+        public Span<T> ToManaged()
+        {
+            ReadOnlySpan<T> managedInner = inner.ToManaged();
+            return MemoryMarshal.CreateSpan(ref MemoryMarshal.GetReference(managedInner), managedInner.Length);
+        }
+
+        public void FreeNative()
+        {
+            inner.FreeNative();
+        }
+    }
+
+    [GenericContiguousCollectionMarshaller]
+    public unsafe ref struct NeverNullSpanMarshaller<T>
+    {
+        private SpanMarshaller<T> inner;
+
+        public NeverNullSpanMarshaller(int sizeOfNativeElement)
+            : this()
+        {
+            inner = new SpanMarshaller<T>(sizeOfNativeElement);
+        }
+
+        public NeverNullSpanMarshaller(Span<T> managed, int sizeOfNativeElement)
+        {
+            inner = new SpanMarshaller<T>(managed, sizeOfNativeElement);
+        }
+
+        public NeverNullSpanMarshaller(Span<T> managed, Span<byte> stackSpace, int sizeOfNativeElement)
+        {
+            inner = new SpanMarshaller<T>(managed, stackSpace, sizeOfNativeElement);
+        }
+
+        /// <summary>
+        /// Stack-alloc threshold set to 256 bytes to enable small spans to be passed on the stack.
+        /// Number kept small to ensure that P/Invokes with a lot of span parameters doesn't
+        /// blow the stack.
+        /// </summary>
+        public const int StackBufferSize = SpanMarshaller<T>.StackBufferSize;
+
+        public Span<T> ManagedValues => inner.ManagedValues;
+
+        public Span<byte> NativeValueStorage
+        {
+            get => inner.NativeValueStorage;
+        }
+
+        public ref byte GetPinnableReference()
+        {
+            if (inner.ManagedValues.Length == 0)
+            {
+                return ref *(byte*)0x1;
```

Comment:
Thoughts on using a sentinel value? `0xa5a5a5a5`? `0x11111111`?

Reply:
I'd be fine using a sentinel. I'll change this to use one.